schema/schema.go

```diff
@@ -119,6 +119,11 @@ type Ignored struct {
 	AutoIncrement bool
 }
 
+type AutoGenCol struct {
+	Name string
+	Type string
```

Comment:
Type as a variable name seems heavily overloaded here. It was not until the end of the PR could I understand what the possible values of "Type" are.

I suggest we use a more prescriptive name and add comments in the code here to clarify.

Reply:
Let me add a comment, makes sense

Reply:
+1 to @Adi8885 

@asthamohta what if we make the variable name itself a bit more expressive, one thing I could think of is `GENERATION_TYPE`. But there could be other possible names too that might work here.